docs/shared/idempotent.rst

```diff
@@ -0,0 +1,3 @@
+.. note::
+
+   The ``uwtools`` drivers are *idempotent*, meaning that actions they successfully complete during one invocation are not repeated in subsequent invocations. For example, an asset like a configuration file will not be recreated when the driver is run again, even if its UW YAML configuration changes. To force recreation, remove the asset(s) in question -- up to and including the entire provisioned run directory -- then re-run the driver, which will recreate any missing assets based on the current configuration.
```

Comment:
Do we want to perhaps put this up front in the base intro rather than updating all of the drivers with the same text?

Reply:
I thought about that, but don't think we can assume that a user who is looking for documentation for a specific driver will have bothered to read the more general docs: I suspect they'll drill down right to the thing they're interested in, and may not read the high-level background.

Reply:
Fair - fortunately we've at least made the text referenceable so we can adjust later. Erring on the side of over-explaining is for the best here too.